app/models/user.rb

```diff
@@ -36,39 +42,37 @@ def liked?(post)
   def name
     "#{firstname} #{surname}"
   end
+  
   def friends
-    friends_array = active_friendships.map { |friendship| friendship.active_friend if friendship.confirmed }
-    friends_array.concat(passive_friendships.map { |friendship| friendship.passive_friend if friendship.confirmed })
-    friends_array.compact
+    active_friends.compact
   end
 
-  def friends_id
-    friends_array = active_friendships.map { |friendship| friendship.active_friend.id }
-    friends_array.concat(passive_friendships.map { |friendship| friendship.passive_friend.id})
-    friends_array.compact
+  def friends_and_requests_id
+    all_friends_and_requests = active_friends.ids.compact.concat(incoming_pending_friends.ids.compact)
+    all_friends_and_requests += all_friends_and_requests.concat(outgoing_pending_friends.ids.compact)
   end
 
-  def pending_requests_to_users
-    active_friendships.map { |friendship| friendship.active_friend if !friendship.confirmed }.compact
-  end
+  # def pending_requests_to_users
+  #   active_friendships.map { |friendship| friendship.active_friend if !friendship.confirmed }.compact
+  # end
 
-  def pending_requests_from_users
-    passive_friendships.map { |friendship| friendship.passive_friend if !friendship.confirmed }.compact
-  end
+  # def pending_requests_from_users
+  #   passive_friendships.map { |friendship| friendship.passive_friend if !friendship.confirmed }.compact
+  # end
 
   def send_request(user)
-    active_friends << user
+    outgoing_pending_friends << user
   end
 
-  def accept_request(user)
-    accept_friendship = passive_friendships.find { |friendship| friendship.passive_friend == user }
-    accept_friendship.confirmed = true
-    accept_friendship.save
-  end
+  # def accept_request(user)
+  #   user_friendship = passive_friendships.find { |friendship| friendship.passive_friend == user }
+  #   accept_friendship.confirmed = true
+  #   accept_friendship.save
+  # end
```

Comment:
**[REQUIRED]**
Why comment this method.
As mentioned by @maelfosso an additional row is created upon friendship confirmation.
```
 def accept_request(user)
    user_friendship = passive_friendships.find { |friendship| friendship.passive_friend == user }
    accept_friendship.confirmed = true
    accept_friendship.save
    Friendship.create!(user_id: accept_friendship.friend_id, friend_id: accept_friendship.user_id, confirmed: true)
  end
```
Having an additional row with reversed user-friend pair upon friendship confirmation is more efficient according to [this article](https://explainextended.com/2009/03/07/selecting-friends) provided in Pathwright.
This was the change being requested by @maelfosso

Reply:
Hi Wathigo, we commented out the method because we have no use for it.
We did implement the reversed user-friend pair.
Can you please check the friendship_controller.rb file, that is where we have the method.